test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Commands/SignCommandTests.cs

```diff
@@ -66,6 +73,39 @@ public void SignCommand_SignPackage()
             }
         }
 
+        [CIOnlyFact]
+        public void SignCommand_SignPackageWithInvalidEkuFails()
+        {
+            // Arrange
+            var testLogger = new TestLogger();
+
+            using (var dir = TestDirectory.Create())
+            using (var zipStream = new SimpleTestPackageContext().CreateAsStream())
+            {
+                var packagePath = Path.Combine(dir, Guid.NewGuid().ToString());
+
+                zipStream.Seek(offset: 0, loc: SeekOrigin.Begin);
+
+                using (var fileStream = File.OpenWrite(packagePath))
+                {
+                    zipStream.CopyTo(fileStream);
+                }
+
+                // Act
+                var result = CommandRunner.Run(
+                    _nugetExePath,
+                    dir,
+                    $"sign {packagePath} -CertificateFingerprint {_trustedTestCertWithInvalidEku.Source.Cert.Thumbprint} -CertificateStoreName {_trustedTestCertWithInvalidEku.StoreName} -CertificateStoreLocation {_trustedTestCertWithInvalidEku.StoreLocation}",
+                    waitForExit: true,
+                    timeOutInMilliseconds: 10000);
+
+                // Assert
+                result.Success.Should().BeFalse();
+                result.AllOutput.Should().Contain(_noTimestamperWarningCode);
+                result.AllOutput.Should().Contain(_invalidEkuError);
```

Comment:
Verify the NU codes here. I would prefer checking codes over the entire localized strings, I feel like this usually just ends up causing pain later when tests have to be changed every time a minor string change is made, and doesn't provide additional benefits that the code or the code + "EKU" wouldn't get.

Reply:
Is this going to happen?